server/common/default_config.py

```diff
@@ -1,129 +1,148 @@
-  # example:
-  #     dataroot:  /path/to/datasets/
-  # or
-  #     dataroot:  s3://bucket/prefix/
-  #
-  # As an alternative, dataroot can be a dictionary, mapping url prefixes to dataroot paths.
-  # example:
-  #     dataroot:
-  #         set1 : /path/to/set1_datasets/
-  #         set2 : /path/to/set2_datasets/
-  # In this case, datasets can be accessed from <server>/set1/<datasetname> or
-  # <server>/set2/<datasetname>.
-
-  dataroot: null
-
-  # The index page when in multi-dataset mode:
-  #   false or null:  this returns a 404 code
-  #   true:  loads a test index page, which links to the datasets that are available in the dataroot
-  #   string/URL:  redirect to this URL:  flask.redirect(config.multi_dataset__index)
-  index: false
-
-  # A list of allowed matrix types.  If an empty list, then all matrix types are allowed
-  allowed_matrix_types: []
-
-  matrix_cache:
-    # The maximum number of datasets that may be opened at one time.  The least recently used dataset
-    # is evicted from the cache first.
-    max_datasets: 5
-
-    # A matrix is automatically removed from the cache after timelimit_s number of seconds.
-    # If timelimit_s is set to None, then there is no time limit.
-    timelimit_s: 30
-
-single_dataset:
-  datapath: null
-  obs_names: null
-  var_names: null
-  about: null
-  title: null
-
-user_annotations:
-  enable: true
-  type: local_file_csv
-  local_file_csv:
-    directory: null
-    file: null
-  ontology:
-    enable: false
-    obo_location: null
-
-embeddings:
-  names : []
-  enable_reembedding: false
-
-diffexp:
-  enable: true
-  lfc_cutoff: 0.01
-  top_n: 10
-  alg_cxg:
-    # The number of threads to use is computed from: min(max_workers, cpu_multipler * cpu_count).
-    # Where cpu_count is determined at runtime.
-    max_workers: 64
-    cpu_multiplier: 4
-
-    # The target number of matrix elements that are evaluated
-    # together in one thread.
-    target_workunit: 16_000_000
-
-data_locator:
-  s3:
-    # s3 region name.
-    #   if true, then the s3 location is automatically determined from the datapath or dataroot.
-    #   if false/null, then do not set.
-    #   if a string, then use that value (e.g. us-east-1).
-    region_name: true
-
-adaptor:
-  cxg_adaptor:
-    # The key/values under tiledb_ctx will be used to initialize the tiledb Context.
-    # If 'vfs.s3.region' is not set, then it will automatically use the setting from
-    # data_locator / s3 / region_name.
-    tiledb_ctx:
-      sm.tile_cache_size:  8589934592
-      sm.num_reader_threads:  32
-
-  anndata_adaptor:
+  app:
+    verbose: false
+    debug: false
+    host: "127.0.0.1"
+    port : null
+    open_browser: false
+    force_https: false
+    flask_secret_key: null
+    generate_cache_control_headers: false
+    server_timing_headers: false
+    csp_directives: null
+
+  multi_dataset:
+    # If dataroot is set, then cellxgene may serve multiple datasets.  This parameter is not
```

Comment:
it seems a bit odd that the single/multi-dataset mode is enabled by a sub-parameter of multi_dataset configuration.

Maybe it would be clearer with a top level server param that sets the mode directly, which would allow a config file to have full configs or both modes, but only one enabled at a time.

Eg,
```
server:
    data_serving_mode: "multi"  # or "single"
```

Or alternatively:
```
server:
    serve_multiple_datasets: true
```

Reply:
It is clunky right now, agreed.  I'll take your suggestion to add a top level server param.

Reply:
I remembered my I did it the way I did now.  
In practice I expected either the multi_dataset OR the single_dataset block to appear, not both.
So, if the user wanted multi_dataset, then only that block would appear in the config file, and "dataroot" would be a required parameter.  

I think if we made a change with that it would be orthogonal to this PR, and we can do it separately if needed.

One improvement through would be to have an "is_single_dataset()" or "is_multi_dataset()" function on the AppConfig object, so that the logic in the rest of the code is simpler.

Reply:
I'm OK with it being a separate PR, but I think (in principle) that flags/modes set implicitly, ie, by inferring from other flags, is generally bad form.  It makes the system hard to understand.  I can live with it for small stuff, but this is a very fundamental mode for the system.   Just making the argument about clarity and comprehensibility of the system.